### ISSUE TITLE
Tables: CombiTimeTable with Akima spline interpolation and borderline slopes

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaStandardLibrary#1039]
**Reported by beutlich on 20 Mar 2013 15:27 UTC**
Consider the following table model with Akima spline interpolation

```modelica
model ProblematicAkima
  parameter Real eps=0.1;
  Modelica.Blocks.Sources.CombiTimeTable t_old(
    table=[0+eps,5;1,3;2,1;3,1;4,1;5,3],
    smoothness=Modelica.Blocks.Types.Smoothness.ContinuousDerivative) annotation(Placement(transformation(extent={{-100,60},{-80,80}})));
end ProblematicAkima;
```

As long as `eps` is small but greater than zero we get similar curves for the interpolated values. For `eps` = 0 there is a behavioral changeover of the interpolated values that suddenly result in values less than one for `time` in (2,3).

Should an alternative variant of the Akima spline interpolation be used that interpolates (for `eps` = 0) linearly between `time` in [1,2] with the drawback of a discontinuous derivative at `time` = 2?

![ProblematicAkima](https://user-images.githubusercontent.com/14896695/62413485-a88ba700-b60f-11e9-87f0-c8a57441b741.png)

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1039

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 26 Mar 2013 11:13 UTC**
This is most likely not the only case where the Akima spline might result in slightly non-intuitive interpolation, since there are no guarantees about monotonic behavior. Therefore, one would have to analyse all possible cases. I would suggest to stick to the pure Akima spline and do not handle the case above specially.

My goal is to make a new call for enhanced table interpolation features once MSL3.2.1 is released. One of the new features should be monotone spline interpolation, such as "Fritsch F.N. and Carlson R.E. (1980): Monotone piecewise cubic interpolation. SIAM J. Numerc. Anal., Vol. 17, No. 2, April 1980, pp. 238-246", which is, e.g., already used in Modelica.Fluid.Utilities.regRoot2 (if table points are monotone, then all interpolated values are also monotone; interpolation is performed with third order polynomials and the first derivative is continuous).

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 26 Mar 2013 19:15 UTC**
This is not really fixed and should be kept open until we have a solution.

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 27 Mar 2013 07:30 UTC**
The Akima spline will not be changed, it will stay as it is also in the future. Therefore, this ticket should be closed. However, the plan is to introduce another type of interpolation in the future with other properties. However, how this is done should be discussed in the MAP-LIB group and the sketch above is just one option. So, the right way would be to open a new ticket for a future extension. However, I would open a new ticket only after the discussion in the group once all planned extensions for the tables are clarified.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 27 Mar 2013 11:35 UTC**
Still this ticket can not be seen as fixed but rather you decided this is going to be a "wontfix"

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 27 Mar 2013 11:36 UTC**
Closing as wontfix since library officer decided to not fix this yet.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 3 Jun 2013 08:09 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 11 May 2015 18:34 UTC**
See #1717 for the enhancement proposed by comment:1.
